snafu/benchmarks/uperf.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Wrapper for running the uperf benchmark. See http://uperf.org/ for more information."""
+from typing import Iterable, List, Tuple, TypedDict
+import re
+import datetime
+from snafu.benchmarks import Benchmark, BenchmarkResult
+from snafu.config import ConfigArgument, check_file
+from snafu.process import sample_process, ProcessSample
+
+
+class RawUperfResult(TypedDict):
+    timestamp: float
+    bytes: int
+    ops: int
+
+
+class UperfConfig(TypedDict, total=False):
+    # Parsed from stdout
+    test_type: str
+    protocol: str
+    message_size: int
+    read_message_size: int
+    num_threads: int
+    duration: int
```

Comment:
I think these are results and not configurations. The current implementation of uperf simply adds this data to the individual datapoints in it's `_json_payload()` so I would consider this part of `RawUperfResult`

Reply:
Got it OK. In ``_json_payload``, these data are labeled as ``config`` (see https://github.com/cloud-bulldozer/benchmark-wrapper/blob/c649d9db9c0dc3dd563b698cad759d74209b2d79/snafu/uperf_wrapper/trigger_uperf.py#L107) which is why I did the same. Didn't realize it was part of the result, I'll make a change.

Reply:
Hmm, looking at benchmark-operator, the variables in question are set as arguments for the benchmark. Check out: https://github.com/cloud-bulldozer/benchmark-operator/blob/master/roles/uperf/templates/workload.yml.j2#L140-L152

I think that there are a couple of different terms that may be getting confused here. Here is how I see it:

* ``UperfConfig``: contains all the variables that were given as arguments to setup the benchmark
* ``RawUperfResult``: contains a statistic that was emitted from the benchmark
* ``UperfStdout``: contains stuff that is parsed from the output of Uperf
* ``UperfResultData``: contains the parsed stdout statistics

I think that instead of adding the variables in question to ``RawUperfResult``, I'll change the names of ``RawUperfResult`` and ``UperfResultData`` to ``RawUperfStat`` and ``UperfStat`` respectively. Thoughts?